### PR TITLE
Compile tests with CUDA

### PR DESCRIPTION
**Pull request by lukasm91**
Note: This does not create specific CUDA tests yet.

### CODE REVIEW

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
I will continue in a new branch from here on

**Comment by lukasm91** [comment]
launch jenkins

File: include/cpp_bindgen/fortran_array_view.hpp [hunk 1]
```diff
@@ -160,18 +169,21 @@ namespace cpp_bindgen {
         enable_if_t<std::is_lvalue_reference<T>::value && std::is_array<remove_reference_t<T>>::value &&
                     std::is_arithmetic<remove_all_extents_t<remove_reference_t<T>>>::value>> : std::true_type {};
 
+    // simplify this when support for CUDA 9.2 is gone
```
Comment (havogt) [inline include/cpp_bindgen/fortran_array_view.hpp]: It can also be simplified once legacy support is removed?

File: jenkins/build.sh [hunk 2]
```diff
@@ -57,8 +57,10 @@ nice make -j8
 ./driver
 
 # test gt legacy mode
-cd ${cwd}
-mkdir -p build_legacy && cd build_legacy
-cmake .. -DCMAKE_INSTALL_PREFIX=${cwd}/install -DCPP_BINDGEN_GT_LEGACY=ON
-nice make -j8 install
-ctest .
+if [[ "$HOST" != kesch* ]]; then
```
Comment (havogt) [inline jenkins/build.sh:67]: Add comment why this is not tested, i.e. CUDA 8...

File: tests/regression/array_gt_legacy/implementation.cpp [hunk 3]
```diff
@@ -10,7 +10,9 @@
 
 #include <array>
 
+#ifndef CPP_BINDGEN_GT_LEGACY
```
Comment (havogt) [inline tests/regression/array_gt_legacy/implementation.cpp]: Probably we should simplify this now (I used a different pattern before):
I think it should just be
`#include <cpp_bindgen/export_gt_legacy.hpp>` without the defines

**Review by havogt — CHANGES_REQUESTED** [review]

**Comment by lukasm91** [comment]
launch jenkins

File: include/cpp_bindgen/fortran_array_view.hpp (hunk 1 above)
Comment (lukasm91) [reply include/cpp_bindgen/fortran_array_view.hpp]: yes

**Review by havogt — APPROVED** [review]

**Pull request merged**